src/core/core.scale.js

```diff
@@ -623,12 +635,10 @@ var Scale = Element.extend({
 	 */
 	handleMargins: function() {
 		var me = this;
-		if (me.margins) {
```

Comment:
This change is fine, but has nothing to do with sampling the ticks. I'd do this in another PR.